model/host/db.go

```diff
@@ -1265,6 +1266,9 @@ func StartingHostsByClient() (map[ClientOptions][]Host, error) {
 		{
 			"$match": bson.M{StatusKey: evergreen.HostStarting},
 		},
+		{
+			"$limit": batchSize,
```

Comment:
should we sort olderst first before limiting so we get a better shot of getting something that's up when we're rate limiting?